pkg/cloudprovider/cloud.go

```diff
@@ -45,6 +45,10 @@ type Interface interface {
 	ProviderName() string
 	// ScrubDNS provides an opportunity for cloud-provider-specific code to process DNS settings for pods.
 	ScrubDNS(nameservers, searches []string) (nsOut, srchOut []string)
+	// KeyManagementService provides a named key management service supported by the cloud. Returns nil
+	// without an error if key management providers are not supported by the cloud. Returns an error if the named provider
+	// is not supported by the cloud.
+	KeyManagementService(name string) (KeyManagementServices, error)
```

Comment:
KeyManagementService (singular) returning KeyManagementServices (plural) reads oddly. also, the nil without error distinction is odd